L1Trigger/TrackFindingTracklet/test/AnalyzerTracklet.cc

```diff
@@ -0,0 +1,247 @@
+    // read in tracklet products
+    Handle<TTTracks> handle;
+    iEvent.getByToken<TTTracks>(edGetToken_, handle);
+    // read in MCTruth
+    const StubAssociation* selection = nullptr;
+    const StubAssociation* reconstructable = nullptr;
+    if (useMCTruth_) {
+      Handle<StubAssociation> handleSelection;
+      iEvent.getByToken<StubAssociation>(edGetTokenSelection_, handleSelection);
+      selection = handleSelection.product();
+      prof_->Fill(9, selection->numTPs());
+      Handle<StubAssociation> handleReconstructable;
+      iEvent.getByToken<StubAssociation>(edGetTokenReconstructable_, handleReconstructable);
+      reconstructable = handleReconstructable.product();
+      for (const auto& p : selection->getTrackingParticleToTTStubsMap())
+        fill(p.first, hisEffTotal_);
+    }
+    //
+    const TTTracks& ttTracks = *handle.product();
+    vector<TTTrackRefs> ttTrackRefsRegions(setup_->numRegions());
+    vector<int> nTTTracksRegions(setup_->numRegions(), 0);
+    for (const TTTrack<Ref_Phase2TrackerDigi_>& ttTrack : ttTracks)
+      nTTTracksRegions[ttTrack.phiSector()]++;
+    for (int region = 0; region < setup_->numRegions(); region++)
+      ttTrackRefsRegions[region].reserve(nTTTracksRegions[region]);
+    int i(0);
+    for (const TTTrack<Ref_Phase2TrackerDigi_>& ttTrack : ttTracks)
+      ttTrackRefsRegions[ttTrack.phiSector()].emplace_back(TTTrackRef(handle, i++));
+    for (int region = 0; region < setup_->numRegions(); region++) {
+      const TTTrackRefs& ttTrackRefs = ttTrackRefsRegions[region];
+      const int nStubs = accumulate(ttTrackRefs.begin(), ttTrackRefs.end(), 0, [](int& sum, const TTTrackRef& ttTrackRef){ return sum += ttTrackRef->getStubRefs().size(); });
+      const int nTracks = ttTrackRefs.size();
+      hisChannel_->Fill(nTracks);
+      profChannel_->Fill(region, nTracks);
+      prof_->Fill(1, nStubs);
+      prof_->Fill(2, nTracks);
+      prof_->Fill(3, 0);
+    }
+    // analyze tracklet products and associate found tracks with reconstrucable TrackingParticles
+    set<TPPtr> tpPtrs;
+    set<TPPtr> tpPtrsSelection;
+    int allMatched(0);
+    vector<vector<TTStubRef>> tracks;
+    tracks.reserve(ttTracks.size());
+    transform(ttTracks.begin(), ttTracks.end(), back_inserter(tracks), [](const TTTrack<Ref_Phase2TrackerDigi_>& ttTrack){ return ttTrack.getStubRefs(); });
+    if (useMCTruth_) {
+      int tmp(0);
+      associate(tracks, selection, tpPtrsSelection, tmp);
+      associate(tracks, reconstructable, tpPtrs, allMatched);
+    }
+    for (const TPPtr& tpPtr : tpPtrsSelection)
+      fill(tpPtr, hisEff_);
+    prof_->Fill(4, allMatched);
+    prof_->Fill(5, ttTracks.size());
+    prof_->Fill(6, tpPtrs.size());
+    prof_->Fill(7, tpPtrsSelection.size());
+    prof_->Fill(8, 0);
+    nEvents_++;
+  }
+
+  void AnalyzerTracklet::endJob() {
+    // effi
+    eff_->SetPassedHistogram(*hisEff_, "f");
+    eff_->SetTotalHistogram (*hisEffTotal_, "f");
```

Comment:
Why is option "f" used here? (i.e. disable histo consistency checks).

Reply:
either the twiki page explaining me how to make efficiency plots told me that this is important or the code where I did copy paste from had this already. I am pretty sure that it has a reason.